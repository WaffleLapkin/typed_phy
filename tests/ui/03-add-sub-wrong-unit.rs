use typed_phy::IntExt;

fn main() {
    let _ = 5.kg() + 10.sqm();
    let _ = 10.m() - 5.mps();
}
