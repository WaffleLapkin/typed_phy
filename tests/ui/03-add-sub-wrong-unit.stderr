error[E0308]: mismatched types
 --> $DIR/03-add-sub-wrong-unit.rs:4:22
  |
4 |     let _ = 5.kg() + 10.sqm();
  |                      ^^^^^^^^ expected struct `typenum::int::Z0`, found struct `typenum::int::PInt`
  |
  = note: expected struct `typed_phy::quantity::Quantity<{integer}, typed_phy::unit::Unit<typed_phy::dimensions::Dimensions<typenum::int::Z0, typenum::int::PInt<typenum::uint::UInt<typenum::uint::UTerm, typenum::bit::B1>>, _, _, _, _, _>>>`
             found struct `typed_phy::quantity::Quantity<{integer}, typed_phy::unit::Unit<typed_phy::dimensions::Dimensions<typenum::int::PInt<typenum::uint::UInt<typenum::uint::UInt<typenum::uint::UTerm, typenum::bit::B1>, typenum::bit::B0>>, typenum::int::Z0, _, _, _, _, _>>>`

error[E0308]: mismatched types
 --> $DIR/03-add-sub-wrong-unit.rs:5:22
  |
5 |     let _ = 10.m() - 5.mps();
  |                      ^^^^^^^ expected struct `typenum::int::Z0`, found struct `typenum::int::NInt`
  |
  = note: expected struct `typed_phy::quantity::Quantity<{integer}, typed_phy::unit::Unit<typed_phy::dimensions::Dimensions<_, _, typenum::int::Z0, _, _, _, _>>>`
             found struct `typed_phy::quantity::Quantity<{integer}, typed_phy::unit::Unit<typed_phy::dimensions::Dimensions<_, _, typenum::int::NInt<typenum::uint::UInt<typenum::uint::UTerm, typenum::bit::B1>>, _, _, _, _>>>`
